pkg/utils/common.go

```diff
@@ -4,12 +4,32 @@ import (
 	"fmt"
 	"strings"
 
+	"golang.org/x/exp/constraints"
 	gwv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
 )
 
 type MapFunc[T any, U any] func(T) U
 type FilterFunc[T any] func(T) bool
 
+func min[T constraints.Ordered](a, b T) T {
+	if a < b {
+		return a
+	}
+	return b
+}
+
+func Chunks[T any](in []T, size int) [][]T {
+	out := [][]T{}
+	if size <= 0 {
+		return out
+	}
+	for low := 0; low < len(in); low += size {
+		high := min(low+size, len(in))
+		out = append(out, in[low:high])
+	}
+	return out
+}
```

Comment:
Also a rule of thumb, especially for generic algorithms, keep them performant, with low overhead.
For example your and mine version use append. Append - will copy underlying items every time when underlying array cannot hold items anymore, so it will allocate new memory, twice bigger than before and copy all items.

So if you know exact size of the slice you want, it's better to preallocate memory and then use indices. It's not a big deal at all for current usage, but it makes difference when you work with thousands of items in the list. Especially not pointer values, but structs that are not cheap to copy.

```go
func Chunks[T any](in []T, size int) [][]T {
	out := make([][]T, len(in)/size+1)
	if size <= 0 {
		return out
	}
	for low := 0; low < len(in); low += size {
		high := min(low+size, len(in))
                chunkId := low / size
		out[chunkId] = in[low:high]
	}
	return out
}
```